config/filter.d/screensharingd.conf

```diff
@@ -0,0 +1,33 @@
+# Fail2Ban configuration file
+#
+# Author: Simon Brown
+#
+# $Revision: 1 $ 
+#
+# Filter for Mac OS X Screen Sharing service
+
+[INCLUDES]
+
+# Read common prefixes. If any customizations available -- read them from
+# common.local
+before = common.conf
+
+
+[Definition]
+
+_daemon = screensharingd
+
+# Option:  failregex
+# Notes.:  regex to match the password failures messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+# Values:  TEXT
+#
+failregex = ^.+ screensharingd.+: Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: DH$
```

Comment:
I admittedly wasn't sure how __prefix worked, and wasn't expecting that it would do the right thing in this case. But running it does work as expected, and matches the result of my previous regex, so I'll definitely change it to match.

Old version:
 ```fail2ban-regex /var/log/system.log  /etc/fail2ban/filter.d/screensharingd.conf 

Running tests
=============

Use   failregex filter file : screensharingd, basedir: /etc/fail2ban
Use         log file : /var/log/system.log
Use         encoding : UTF-8


Results
=======

Failregex: 22 total
|-  #) [# of hits] regular expression
|   1) [22] ^.+ screensharingd.+: Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: DH$
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [3820] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
|  [6] Year(?P<_sep>[-/.])Month(?P=_sep)Day 24hour:Minute:Second(?:,Microseconds)?
`-

Lines: 4013 lines, 0 ignored, 22 matched, 3991 missed [processed in 0.28 sec]
Missed line(s): too many to print.  Use --print-all-missed to print all 3991 lines```

New Version:

```fail2ban-regex /var/log/system.log  /etc/fail2ban/filter.d/screensharingd.conf 

Running tests
=============

Use   failregex filter file : screensharingd, basedir: /etc/fail2ban
Use         log file : /var/log/system.log
Use         encoding : UTF-8


Results
=======

Failregex: 22 total
|-  #) [# of hits] regular expression
|   1) [22] ^\s*(<[^.]+\.[^.]+>)?\s*(?:\S+ )?(?:kernel: \[ *\d+\.\d+\] )?(?:@vserver_\S+ )?(?:(?:\[\d+\])?:\s+[\[\(]?screensharingd(?:\(\S+\))?[\]\)]?:?|[\[\(]?screensharingd(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)?\s(?:\[ID \d+ \S+\])?\s*Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: \w+$
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [3820] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
|  [6] Year(?P<_sep>[-/.])Month(?P=_sep)Day 24hour:Minute:Second(?:,Microseconds)?
`-

Lines: 4013 lines, 0 ignored, 22 matched, 3991 missed [processed in 0.30 sec]
Missed line(s): too many to print.  Use --print-all-missed to print all 3991 lines```